### PR TITLE
Fix AVR ws2812 when ADDRESS_BASE is non zero

### PR DESCRIPTION
**Pull request by zvecr**
## Description

Prerequisite for #8598.

Aligned calls with logic within `config_common.h`.
Should also allow 32a support too (though I have not tested this).

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: drivers/avr/ws2812.c [hunk 1]
```diff
@@ -20,33 +20,35 @@
...
 
-void inline ws2812_setleds_pin(LED_TYPE *ledarray, uint16_t leds, uint8_t pinmask) {
-    // new universal format (DDR)
-    _SFR_IO8((RGB_DI_PIN >> 4) + 1) |= pinmask;
+void ws2812_setleds_pin(LED_TYPE *ledarray, uint16_t number_of_leds, uint8_t pin) {
+    DDRx_ADDRESS(RGB_DI_PIN) |= pinmask(pin);
 
-    ws2812_sendarray_mask((uint8_t *)ledarray, leds * sizeof(LED_TYPE), pinmask);
+    uint8_t masklo = ~(pinmask(pin)) & PORTB;
+    uint8_t maskhi = pinmask(pin) | PORTB;
```
Comment (fauxpark) [inline drivers/avr/ws2812.c:49]: `PORTx_ADDRESS`?

Comment (zvecr) [reply drivers/avr/ws2812.c:49]: fuuuu thats what i get for not checking my diffs.

**Comment by drashna** [comment]
Can confirm that it works on the Ergodox EZ, without any immediate issues. LEDs work, animations work, layer changing doesn't trigger any issues, etc.  Looks good.

**Comment by zvecr** [comment]
@drashna Cheers for the testing!

**Pull request merged**